fileio/simpleio.go

```diff
@@ -20,9 +20,9 @@ const (
 // and a pointer to os.File for closeure when reading is complete.
 type SimpleReader struct {
 	*bufio.Reader
-	file   *os.File
+	close  func() error
```

Comment:
When using this struct, what is the difference between calling .Close(), like at the end of ReadLine below, or calling the .close() function on this line?

Reply:
I'm not too sure if there are differences other than small style preferences. I didn't want to keep *os.File around only to close the reader, so I defined a closure function to run the same behavior as os.File.close.
If i have `reader := NewSimpleReader(filename)`
I wanted to run `defer reader.Close()` or `reader.Close()` and close it at the end, without returning the error and mimic the close function in easyio reader.